code/modules/antagonists/cult/team_cult.dm

```diff
@@ -0,0 +1,485 @@
+	RegisterSignal(new_member.current, COMSIG_PARENT_QDELETING, PROC_REF(remove_member))
+
+/datum/team/cult/handle_removing_member(datum/mind/member)
+	. = ..()
+	UnregisterSignal(member.current, COMSIG_MOB_STATCHANGE)
+	UnregisterSignal(member.current, COMSIG_PARENT_QDELETING)
+	check_cult_size()
+
+
+/datum/team/cult/on_round_end()
+	var/list/endtext = list()
+	endtext += "<br><b>The cultists' objectives were:</b>"
+	for(var/datum/objective/obj in objective_holder.get_objectives())
+		endtext += "<br>[obj.explanation_text] - "
+		if(!obj.check_completion())
+			endtext += "<font color='red'>Fail.</font>"
+		else
+			endtext += "<font color='green'><B>Success!</B></font>"
+
+	to_chat(world, endtext.Join(""))
+
+/datum/team/cult/proc/add_cult_immunity(mob/living/target)
+	ADD_TRAIT(target, TRAIT_CULT_IMMUNITY, CULT_TRAIT)
+	addtimer(CALLBACK(src, PROC_REF(remove_cult_immunity), target), 1 MINUTES)
+
+/datum/team/cult/proc/remove_cult_immunity(mob/living/target)
+	REMOVE_TRAIT(target, TRAIT_CULT_IMMUNITY, CULT_TRAIT)
+
+/**
+ * Makes sure that the signal stays on the correct body when a cultist changes bodies
+ */
+/datum/team/cult/proc/cult_body_transfer(old_body, new_body)
+	UnregisterSignal(old_body, COMSIG_MOB_STATCHANGE)
+	RegisterSignal(new_body, COMSIG_MOB_STATCHANGE, PROC_REF(cultist_stat_change))
+
+/**
+  * Returns the current number of cultists and constructs.
+  *
+  * Returns the number of cultists and constructs in a list ([1] = Cultists, [2] = Constructs), or as one combined number.
+  *
+  * * separate - Should the number be returned as a list with two separate values (Humans and Constructs) or as one number.
+  */
+/datum/team/cult/proc/get_cultists(separate = FALSE)
+	var/cultists = 0
+	var/constructs = 0
+	for(var/datum/mind/M as anything in members)
+		if(isnull(M))
+			stack_trace("Found a null mind in /datum/team/cult's members.")
+			continue
+		if(isnull(M.current))
+			stack_trace("Found a mind with no body in /datum/team/cult's members.")
+			continue
+		if(QDELETED(M) || M.current.stat == DEAD)
+			continue
+		if(ishuman(M.current) && !M.current.has_status_effect(STATUS_EFFECT_SUMMONEDGHOST))
+			cultists++
+		else if(isconstruct(M.current))
+			constructs++
+	if(separate)
+		return list(cultists, constructs)
+	return cultists + constructs
+
+/datum/team/cult/proc/cultist_stat_change(mob/target_cultist, new_stat, old_stat)
+	if(new_stat == old_stat) // huh, how? whatever, we ignore it
+		return
+	if(new_stat != DEAD && old_stat != DEAD)
+		return // switching between alive and unconcious
+	// switching between dead and alive/unconcious
+	check_cult_size()
+
+/datum/team/cult/proc/check_cult_size()
+	var/cult_players = get_cultists()
+
+	if(cult_ascendant)
+		// The cult only falls if below 1/2 of the rising, usually pretty low. e.g. 5% on highpop, 10% on lowpop
+		if(cult_players < (rise_number / 2))
+			cult_fall()
+		return
+
+	if((cult_players >= rise_number) && !cult_risen)
```

Comment:
```suggestion
	if(cult_players >= rise_number && !cult_risen)
```

Reply:
I think this is important to keep in order to read order of operations